tracing-opentelemetry/src/subscriber.rs

```diff
@@ -555,6 +555,17 @@ where
                     builder.status_code = Some(otel::StatusCode::Error);
                 }
 
+                let builder_attrs = builder.attributes.get_or_insert(Vec::new());
```

Comment:
Think it would be a good idea to allow this behavior to be configured similar to [`with_tracked_inactivity`](https://github.com/tokio-rs/tracing/blob/master/tracing-opentelemetry/src/subscriber.rs#L339) as this will introduce a small amount of overhead for the additional allocations to all instrumented events?

Reply:
+1 for making this configurable.

Reply:
we may also want to consider pre-allocating the `Vec` using `Vec::with_capacity` based on the number of metadata fields that are `Some`...but this is _probably_ a premature optimization, especially since I think the first allocation from growing the vec should fit everything. probably not necessary.